### PR TITLE
Add support for off-the-record (incognito) mode

### PR DESCRIPTION
**Pull request by svillar**
This commit adds a new browser context to WolvicMainParts that could be used for off-the-record browsing. Tabs got a new parametter on creation that specifies whether or not to use an off-the-record browsing context.

### CODE REVIEW

**Comment by svillar** [comment]
Wolvic part implemented here https://github.com/Igalia/wolvic/pull/986

File: wolvic/browser/tab_jni.cc [hunk 1]
```diff
@@ -18,14 +18,14 @@ using web_contents_delegate_android::WebContentsDelegateAndroid;
...
+  ScopedJavaLocalRef<jobject> JNI_Tab_CreateWebContents(JNIEnv* env, jboolean is_off_the_record) {
   // TODO(wolvic-chromium#6): Consider handling browser profiles.
   auto* browser_client = content::WolvicContentBrowserClient::Get();
-  CHECK(browser_client->GetBrowserContext() != nullptr);
+  CHECK(browser_client->browser_context() != nullptr);
 
   std::unique_ptr<WebContents> web_contents =
       WebContents::Create(content::WebContents::CreateParams(
-          browser_client->GetBrowserContext()));
+							     is_off_the_record ? browser_client->off_the_record_browser_context() : browser_client->browser_context()));
```
Comment (jdapena) [inline wolvic/browser/tab_jni.cc]: :m line length seems wrong? Could you run clang-format?

**Comment by svillar** [comment]
Can be tested using [this demo](https://fingerprint.com/demo) for example

**Review by jdapena — APPROVED** [review]
Patch LGTM. Though, it looks like it needs to get a clang-format pass?

File: wolvic/browser/tab_jni.cc (hunk 1 above)
Comment (svillar) [reply wolvic/browser/tab_jni.cc]: Sure

**Pull request merged**